lib/components/user/notification-prefs-pane.js

```diff
@@ -218,68 +155,244 @@ class NotificationPrefsPane extends Component {
+              validateOnChange
+              validationSchema={this.phoneValidationSchema}
+              initialValues={initialFormikValues}
+            >
+              {
+                // Pass Formik props to the component rendered so Formik can manage its validation.
+                // (The validation for this component is independent of the validation set in UserAccountScreen.)
+                innerProps => {
+                  return (
+                    <PhoneNumberEditor
+                      {...innerProps}
+                      initialPhoneNumber={phoneNumber}
+                      initialPhoneNumberVerified={isPhoneNumberVerified}
+                      onRequestCode={onRequestPhoneVerificationCode}
+                      onSubmitCode={onSendPhoneVerificationCode}
+                      onVerified={this._handleOnPhoneVerified}
+                      phoneFormatOptions={phoneFormatOptions}
+                    />
+                  )
+                }
+              }
+            </Formik>
           )}
         </Details>
+      </div>
+    )
+  }
+}
 
-        {/* The dialog prompt for validation code. */}
-        <Modal onHide={this._handlePhoneValidationCancel} show={isVerificationPending}>
-          <Modal.Header>
-            <Modal.Title>Enter verification code</Modal.Title>
-          </Modal.Header>
+export default NotificationPrefsPane
 
-          <Modal.Body>
-            <p>
-              Please check the SMS messaging app on your mobile phone
-              for a text message with a verification code, and enter the code below.
-            </p>
+/**
+ * Sub-component that handles phone number and validation code editing and validation intricacies.
+ */
+class PhoneNumberEditor extends Component {
+  constructor (props) {
+    super(props)
+
+    const { initialPhoneNumber, initialPhoneNumberVerified } = props
+    this.state = {
+      // Holds the last known verified phone number.
+      currentNumber: formatPhoneNumber(initialPhoneNumber, props.phoneFormatOptions),
+
+      // The verified status of the above number.
+      currentNumberVerified: initialPhoneNumberVerified,
+
+      // If true, phone number is being edited.
+      // For new users, render component in editing state.
+      isEditing: isBlank(initialPhoneNumber),
+
+      // If true, a phone verification request has been sent and the UI is awaiting the user sending the code.
+      isPending: false
+    }
+  }
+
+  _handleEditNumber = () => {
+    this.setState({
+      isEditing: true,
+      isPending: false
+    })
+  }
+
+  _handleCancelEditNumber = () => {
+    this.setState({
+      isEditing: false,
+      isPending: false
+    })
+  }
+
+  _handlePhoneNumberKeyDown = e => {
+    if (e.keyCode === 13) {
+      // On the user pressing enter (keyCode 13) on the phone number field,
+      // prevent form submission and request the code.
+      e.preventDefault()
+      this._handleRequestCode()
```

Comment:
I'm not sure this (handling Enter keydown) is necessary.

Reply:
I think it is necessary for users who have keyboard "Enter" reflexes.

Reply:
Let's remove these handlers when removing the Save/Cancel buttons for the account page.